src/tests/nativeaot/SmokeTests/PInvoke/PInvoke.cs

```diff
@@ -1053,6 +1105,87 @@ override protected bool ReleaseHandle()
     [InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
     public interface IComInterface
     {
+        void DoWork();
+    }
+
+    public class ComObject : IComInterface
+    {
+        public void DoWork()
+        {
+            Console.WriteLine("Do work called.");
+        }
+    }
+
+    internal unsafe class SimpleComWrapper : ComWrappers
+    {
+        static ComInterfaceEntry* wrapperEntry;
+
+        internal static Guid IComInterface_GUID = new Guid("D6DD68D1-86FD-4332-8666-9ABEDEA2D24C");
+
+        static SimpleComWrapper()
+        {
+            GetIUnknownImpl(out IntPtr fpQueryInteface, out IntPtr fpAddRef, out IntPtr fpRelease);
+
+            var vtbl = new IComInterfaceVtbl()
+            {
+                IUnknownImpl = new IUnknownVtbl()
+                {
+                    QueryInterface = fpQueryInteface,
+                    AddRef = fpAddRef,
+                    Release = fpRelease
+                },
+                DoWork = &IComInterfaceProxy.DoWork,
+            };
+            var vtblRaw = RuntimeHelpers.AllocateTypeAssociatedMemory(typeof(IComInterfaceVtbl), sizeof(IComInterfaceVtbl));
+            Marshal.StructureToPtr(vtbl, vtblRaw, false);
```

Comment:
```suggestion
            IntPtr* vtblRaw = (IntPtr*)RuntimeHelpers.AllocateTypeAssociatedMemory(typeof(IComInterfaceVtbl), sizeof(IComInterfaceVtbl));
            GetIUnknownImpl(out vtbl[0], out out vtbl[1], out out vtbl[2]);
            vtbl[3] = (IntPtr)(delegate* unmanaged<IntPtr, void>)&DoWork
```
I would avoid using `Marshal.StructureToPtr` here.